--- OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/gta/CMakeLists.txt.orig	2020-01-31 12:03:07.000000000 +0100
+++ OpenSceneGraph-OpenSceneGraph-3.6.5/src/osgPlugins/gta/CMakeLists.txt	2023-09-07 21:44:21.308684691 +0200
@@ -2,7 +2,7 @@ INCLUDE_DIRECTORIES( ${GTA_INCLUDE_DIRS}
 
 SET(TARGET_SRC ReaderWriterGTA.cpp )
 
-SET(TARGET_LIBRARIES_VARS GTA_LIBRARY)
+SET(TARGET_LIBRARIES_VARS GTA_LIBRARIES)
 
 #### end var setup  ###
 SETUP_PLUGIN(gta)
