--- OpenSceneGraph-3.0.1/src/osgPlugins/xine/video_out_rgb.c.orig	2012-05-20 11:31:15.157158848 +0200
+++ OpenSceneGraph-3.0.1/src/osgPlugins/xine/video_out_rgb.c	2012-05-20 11:38:48.992055552 +0200
@@ -2769,8 +2769,8 @@ init_class(xine_t* xine, void* vo_visual
 	clear(rgb_class, sizeof(rgbout_class_t));
 
 	rgb_class->driver_class.open_plugin     = open_plugin;
-	rgb_class->driver_class.get_identifier  = get_identifier;
-	rgb_class->driver_class.get_description = get_description;
+	rgb_class->driver_class.identifier  = get_identifier;
+	rgb_class->driver_class.description = get_description;
 	rgb_class->driver_class.dispose         = dispose_class;
 
 	return(rgb_class);
